src/language-js/print/call-arguments.js

```diff
@@ -261,7 +261,12 @@ function shouldGroupFirstArg(args) {
     secondArg.type !== "FunctionExpression" &&
     secondArg.type !== "ArrowFunctionExpression" &&
     secondArg.type !== "ConditionalExpression" &&
-    !couldGroupArg(secondArg)
+    // A hack to fix most manifestations of
+    // https://github.com/prettier/prettier/issues/2456
+    // https://github.com/prettier/prettier/issues/5172
+    // A proper (printWidth-aware) fix for those would require a complex change in the doc printer.
+    !secondArg?.left?.left &&
```

Comment:
Because this  check  didn't pass?

**Prettier pr-13341**
[Playground link](https://deploy-preview-13341--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAucBDANugFAMwK5RgwCW0ABFgJRnAA6UZZATnDHkwwIwDc9AvgBoyAEgBuqJmQA+U+oyxiJZAPzK5I8ZNXrFWtQw1LtB3Sv2NTxi5rM6bVsrRDwAzjCeVKvKPTAZs+IQk5FQ06ixsHGQ8-EKW5hTxdkYJSSb2qRnJetm26SnqTq7uIJ7cIAIgEAAOwVAuyKASTBAA7gAKEggNKBitqACeDZUARkyoYADWrADKqAC2cAAyxFBwyDgYLnCj41Oz1ROrAObIMEx4OyBw8yNwACb3D0uoUMd4qMdwAGIQTPOoGAkN7IECoPAwCAVEAACxg83QAHUYcRXIcwHAZt1UcRRKiBqCwC5hiBVtsmDB2uNjgCNlsrgArFwADxmJ3QcAAingIPA6ehtpVDkxyaCRqg7uhodUmKsYIjiPcYDDkAAOAAMQpa20R42qoJlcHJonWlQAjjz4FSaj0wS4ALRrB4PaEsC3EFhUz60pCbAVXbbzYhnC4B9lcy3rX30yowCUKpUqpAAJlj42I6BOAGEIPMfWDMNC8NsACoSnp+wUgUSXACSUCesBmYFltQAgg2ZjABhz+ds+HwgA)
<!-- prettier-ignore -->
```sh
--parser babel
```

**Input:**
<!-- prettier-ignore -->
```jsx
call(function () {
  return 1;
}, $var ||
  ($var ??
  $var ??
  $var ??
  $var ??
  $var ??
  $var ??
  $var ??
  $var ??
  "test"));

call(function () {
  return 1;
}, $var ??
  ($var ??
  $var ??
  $var ??
  $var ??
  $var ??
  $var ??
  $var ??
  $var ??
  "test"));
```

**Output:**
<!-- prettier-ignore -->
```jsx
call(function () {
  return 1;
}, $var || ($var ?? $var ?? $var ?? $var ?? $var ?? $var ?? $var ?? $var ?? "test"));

call(
  function () {
    return 1;
  },
  $var ??
    $var ??
    $var ??
    $var ??
    $var ??
    $var ??
    $var ??
    $var ??
    $var ??
    "test",
);

```

Reply:
> Because this check didn't pass?

Yep. It's a hack, sure, but it does the trick for most cases.
What would you suggest? Should we add ` && !secondArg?.right?.right`?
I decided not to touch https://github.com/prettier/prettier/issues/12892 and https://github.com/prettier/prettier/issues/4401 for now.

Reply:
Maybe just check if it's LogicalExpression chain for now? Unless you mean to match other types with `left`.

Reply:
`secondArg?.right?.right` will match `a = b = c`

Reply:
> `secondArg?.right?.right` will match `a = b = c`

That would be perfectly okay.

> Unless you mean to match other types with left.

Yes, I mean to match binary expression that are more complex than this:
```js
setTimeout(() => {
   // ...
}, 40 * rate);
```

Reply:
We have full list of those nodes have `left`, `right`, you can add more tests for them.

https://github.com/prettier/prettier/blob/3cae07c9df165fe520423dbf42d6d32145003525/tests/unit/__snapshots__/visitor-keys.js.snap#L3